controller/konnect/watch_credentialbasicauth.go

```diff
@@ -0,0 +1,209 @@
+package konnect
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	operatorerrors "github.com/kong/gateway-operator/internal/errors"
+
+	configurationv1 "github.com/kong/kubernetes-configuration/api/configuration/v1"
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+// TODO(pmalek): this can be extracted and used in reconciler.go
+// as every Konnect entity will have a reference to the KonnectAPIAuthConfiguration.
+// This would require:
+// - mapping function from non List types to List types
+// - a function on each Konnect entity type to get the API Auth configuration
+//   reference from the object
+// - lists have their items stored in Items field, not returned via a method
+
+// CredentialBasicAuthReconciliationWatchOptions returns the watch options for
+// the CredentialBasicAuth.
+func CredentialBasicAuthReconciliationWatchOptions(
+	cl client.Client,
+) []func(*ctrl.Builder) *ctrl.Builder {
+	return []func(*ctrl.Builder) *ctrl.Builder{
+		func(b *ctrl.Builder) *ctrl.Builder {
```

Comment:
Shouldn't we enqueue credentials requests also on the owning `KongConsumer` changes?

Reply:
I'm curious if that's actually necessary 🤔 

The only thing that would be relevant for the credentials here would be the consumer ID or its CP ID which currently cannot change because we disallow migrations between CPs.

Do you see any particular reason that we should enqueue credentials on consumer changes?

Reply:
When the `KongConsumer` is deleted, it won't automatically trigger events for the credentials. In the case of `cascade=foreground` deletion that will make the process wait until the credential gets reconciled again on its fixed updated interval. For the `background` it will also make the final deletion of a credential postponed until the fixed update reconciliation is triggered.

Reply:
I somehow forgot about the deletion. You're right.

4ce45b527d31e19f2e4f7c07698bd373ace5a7b9 should handle that (also added checking this to the test)